### DISCUSSION TITLE
Provisioning and machine inventory

### DISCUSSION THREAD

**Discussion by dantefromhell** [dep-sys/nix-dabei#8] [General]
Issue #2 defines the process of creating & installing a VM, but I'm wondering where the inventory data that describes machine data & state should stored in?

Should nix-dabei have preference for that or describe a data-format instead?
After all, one needs at least a name + sizing information (but possibly more) to create a VM.

**Comment by phaer** [comment]
What state are you thinking about exactly? I am not sure what you mean by "sizing information" as hcloud machines, as mentioned in #2, are technically virtual machines but in my understanding things like disk sizes are kinda out of scope for  `nix-dabei`. You create a VM via terraform or hcloud cli and then provision it via nix-dabei.

Things like hostname, network config, disk name (for partitioning), etc would be gathered in the rescue system and passed to the kexec'd environment via json. In the kernel cmdline as default, but it would be easy enough to allow users to just pass an URL to a JSON file as well. 
Once the target system is provisioned, there's no state needed besides what's already in the system closure, or am I missing something?

**Comment by dantefromhell** [comment]
> Once the target system is provisioned, there's no state needed besides what's already in the system closure, or am I missing something?

Good question, I need to think this through/ experiment a bit more.

> Things like hostname, network config, disk name (for partitioning), etc would be gathered in the rescue system and passed to the kexec'd environment via json. In the kernel cmdline as default, but it would be easy enough to allow users to just pass an URL to a JSON file as well.

I do prefer the idea of a JSON file over kernel commandline. In case one needs to transfer credentials to the auto-installer (private git repo or disc encryption keys come to my mind) these can leak into log files because the kernel command line is included in `dmesg` and `journalctl` output.
